main/main.go

```diff
@@ -0,0 +1,226 @@
+package main
+
+import (
+	"bufio"
+	"github.com/stackrox/CvesPusher/utils"
+	"io/ioutil"
+
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/clair/pkg/commonerr"
+	"github.com/coreos/clair/pkg/httputil"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	dataFeedURL     string = "https://nvd.nist.gov/feeds/json/cve/1.0/nvdcve-1.0-%s.json.gz"
+	dataFeedMetaURL string = "https://nvd.nist.gov/feeds/json/cve/1.0/nvdcve-1.0-%s.meta"
+	logDataFeedName string = "data feed name"
+	filePrefix      string = "NVD"
+	kubernetes      string = "k8s"
+	istio           string = "istio"
+	fileExt         string = ".json"
+	cves            string = "CVEs"
+)
+
+type appender struct {
+	localPath      string
+	dataFeedHashes map[string]string
+	metadata       map[string]NVDMetadata
+}
+
+type NVDMetadata struct {
+	CVSSv2 NVDmetadataCVSSv2
+	CVSSv3 NVDmetadataCVSSv3
+}
+
+type NVDmetadataCVSSv2 struct {
+	PublishedDateTime string
+	Vectors           string
+	Score             float64
+}
+
+type NVDmetadataCVSSv3 struct {
+	Vectors             string
+	Score               float64
+	ExploitabilityScore float64
+	ImpactScore         float64
+}
+
+func main() {
```

Comment:
It's preferable to use a pattern like this for your main function: https://github.com/stackrox/circleci-notifier/blob/master/main.go#L26

That way, you can just return an error always, and only `os.Exit(1)` in one place.